src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -2852,17 +2852,19 @@ void AutomationInstrumentClipView::setParameterAutomationValue(ModelStackWithAut
 	interpolationAfter = getNodeInterpolation(modelStack, squareStart, false);
 
 	//create a node to the left with the current interpolation status
-	if ((squareStart - kParamNodeWidth) >= 0) {
-		int32_t currentValue = modelStack->autoParam->getValuePossiblyAtPos(squareStart - kParamNodeWidth, modelStack);
-		modelStack->autoParam->setValuePossiblyForRegion(currentValue, modelStack, squareStart - kParamNodeWidth,
+	int32_t squareNodeLeftStart = squareStart - kParamNodeWidth;
+	if (squareNodeLeftStart >= 0) {
+		int32_t currentValue = modelStack->autoParam->getValuePossiblyAtPos(squareNodeLeftStart, modelStack);
+		modelStack->autoParam->setValuePossiblyForRegion(currentValue, modelStack, squareNodeLeftStart,
 		                                                 kParamNodeWidth);
 	}
 
```

Comment:
For the node to the right, I renamed the previous variable from squareRightEdge to squareNodeRightStart

also, my previous math for the right adjacent node was wrong. Where it said "squareStart + squareWidth" before, it should've said "squareStart + kParamNodeWidth."